### PR TITLE
feat: 🎸 add type check helper methods to target model

### PR DESCRIPTION
**Pull request by DhariniJeeva**
This pr adds TCP and SSH type check methods to the target model. This is required to support both the target types.

### CODE REVIEW

File: addons/api/tests/unit/models/target-test.js [hunk 1]
```diff
@@ -384,4 +384,24 @@ module('Unit | Model | target', function (hooks) {
     const model = store.peekRecord('target', '123abc');
     await model.removeCredentialSource('2');
   });
+
+  test('it has isSSH property and returns the expected values', async function (assert) {
```
Comment (cameronperera) [inline addons/api/tests/unit/models/target-test.js:388]:
```suggestion
  test('it has isSSH property and returns the expected values', function (assert) {
```
It does not look like we need this to be async.

File: addons/api/tests/unit/models/target-test.js [hunk 2]
```diff
@@ -384,4 +384,24 @@ module('Unit | Model | target', function (hooks) {
...
+    assert.expect(2);
+    const store = this.owner.lookup('service:store');
+    const modelSSH = store.createRecord('target', {
+      type: 'ssh',
+    });
+    assert.true(modelSSH.isSSH);
+    assert.false(modelSSH.isTCP);
+  });
+
+  test('it has isTCP property and returns the expected values', async function (assert) {
```
Comment (cameronperera) [inline addons/api/tests/unit/models/target-test.js:398]:
```suggestion
  test('it has isTCP property and returns the expected values', function (assert) {
```
Same as other comment.

**Review by cameronperera — CHANGES_REQUESTED** [review]

File: addons/api/tests/unit/models/target-test.js (hunk 1 above)
Comment (DhariniJeeva) [reply addons/api/tests/unit/models/target-test.js]: good catch! updated!

**Review by cameronperera — APPROVED** [review]

File: addons/api/addon/models/target.js [hunk 3]
```diff
@@ -207,4 +207,20 @@ export default class TargetModel extends GeneratedTargetModel {
   removeCredentialSource(credentialLibraryID, options) {
     return this.removeCredentialSources([credentialLibraryID], options);
   }
+
+  /**
+   * True if the target type is tcp.
+   * @type {boolean}
+   */
+  get isTCP() {
```
Comment (calcaide) [inline addons/api/addon/models/target.js:215]: Curious, where we will use this? within the targets list?

**Review by calcaide — APPROVED** [review]
LGTM, left a question

File: addons/api/addon/models/target.js (hunk 3 above)
Comment (DhariniJeeva) [reply addons/api/addon/models/target.js:215]: yes! that is right

**Pull request merged**